### PR TITLE
Toolkit neutral menus part 1

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: *(direct link to the issue)*

A part of the work needed to extract a toolkit-neutral facade library for menus and menu bars.

More commits to come, but these preliminary commits are stable enough to invite review now.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/tracks/playabletrack/wavetrack/ui/SpectrumView.cpp [hunk 1]
```diff
@@ -1035,11 +1035,11 @@ PopupMenuTable::AttachedItem sAttachment{
                   (wxCommandEventFunction)
                      (&SpectrogramSettingsHandler::OnSpectrogramSettings),
                   SpectrogramSettingsHandler::Instance(),
-                  []( PopupMenuHandler &handler, wxMenu &menu, int id ){
+                  []{
```
Comment (ksoze95) [inline src/tracks/playabletrack/wavetrack/ui/SpectrumView.cpp:1038]: Does this lambda need a return type defined?

**Comment by Paul-Licameli** [comment]
I have rebased and force pushed, only to resolve one small conflict in CommandManager.h at the commit "Use BasicMenu::Item::Text in interface of CommandManager".  Nothing else changed.

File: src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp [hunk 2]
```diff
@@ -1143,21 +1132,24 @@ BEGIN_POPUP_MENU(WaveColorMenuTable)
...
-      GetWaveColorStr(0), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument2", OnInstrument2ID,
-      GetWaveColorStr(1), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument3", OnInstrument3ID,
-      GetWaveColorStr(2), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument4", OnInstrument4ID,
-      GetWaveColorStr(3), POPUP_MENU_FN( OnWaveColorChange ), fn );
+   for (size_t ii = 0; ii < 3; ++ii) {
+      AppendRadioItem(
+         wxString::Format("Instrument%d", ii),
```
Comment (ksoze95) [inline src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp:1137]: In the original the popup menu items starting from Instrument1 to Instrument4.  This has been refactored to have one less menu item and start from Instrument0, is this intentional?

Comment (Paul-Licameli) [reply src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp:1137]: Aha, good catch.  Unintended change of menu labels.  I should insert `+ 1`

Comment (Paul-Licameli) [reply src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp]: Rebased onto recent master; the +1 fix is now squashed into the last commit

**Comment by Paul-Licameli** [comment]
Latest force-push is only a rebasing onto recent master, no changes

**Comment by Paul-Licameli** [comment]
Sorry for the force-push -- but there were non-trivial conflicts to resolve in WaveTrackControls.cpp, after the merger of less-dithering.

File: src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp [hunk 3]
```diff
@@ -1143,21 +1132,24 @@ BEGIN_POPUP_MENU(WaveColorMenuTable)
...
 
-   AppendRadioItem( "Instrument1", OnInstrument1ID,
-      GetWaveColorStr(0), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument2", OnInstrument2ID,
-      GetWaveColorStr(1), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument3", OnInstrument3ID,
-      GetWaveColorStr(2), POPUP_MENU_FN( OnWaveColorChange ), fn );
-   AppendRadioItem( "Instrument4", OnInstrument4ID,
-      GetWaveColorStr(3), POPUP_MENU_FN( OnWaveColorChange ), fn );
+   for (size_t ii = 0; ii < 3; ++ii) {
```
Comment (ksoze95) [inline src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp:1135]: Terminating condition be changed to ii < 4.  It currently is missing the "Instrument 4" menu item.

**Pull request closed without merging**